lib/process_settings/file_monitor.rb

```diff
@@ -24,6 +25,26 @@ def initialize(file_path, logger:, environment: nil)
       start_internal(enable_listen_thread?(environment))
     end
 
+    def start_watchdog_thread(file_path)
+      @watchdog_thread and raise ArgumentError, "watchdog thread already running!"
+      @watchdog_thread = Thread.new do
+        watchdog = ProcessSettings::Watchdog.new(file_path)
+        loop do
+          ExceptionHandling.ensure_safe("ProcessSettings::Watchdog thread") do
```

Comment:
@ColinDKelley How do I go about flattening the [ensure_safe](https://github.com/Invoca/exception_handling/blob/master/lib/exception_handling.rb#L310-L315) method from ExceptionHandling? Would I call `Logger.new` or a `warn` instead of trying to extract all of the calls made in the `log_error` method? Is that the right idea?

Reply:
This object already has a `logger`, so no need to create one with `Logger.new`, but  yes, the flattened version of `ensure_safe` would call `logger.error` itself:
```
loop do
  sleep(1.minute)
  watchdog.check
rescue => ex
  logger.error("ProcessSettings::Watchdog thread: #{ex.class.name}: #{ex.message}")
end
```